pkg/contracts/test/setups/OsmoticPoolSetup.sol

```diff
@@ -0,0 +1,127 @@
+
+import {OsmoticController} from "../../src/OsmoticController.sol";
+import {OsmoticPool, ProjectSupport} from "../../src/OsmoticPool.sol";
+import {IProjectList} from "../../src/interfaces/IProjectList.sol";
+import {ProjectRegistry} from "../../src/projects/ProjectRegistry.sol";
+
+abstract contract OsmoticPoolSetup is BaseSetup {
+    uint256 constant MAX_ACTIVE_PROJECTS = 25;
+
+    MimeToken mimeToken;
+    OsmoticController controller;
+    OsmoticPool pool;
+    ProjectRegistry projectRegistry;
+    IProjectList projectList;
+
+    address poolOwner;
+
+    uint256[] projectIds;
+
+    function setUp() public virtual override {
+        super.setUp();
+
+        (, address projectRegistryAddress) = createProjectRegistry(VERSION);
+        projectRegistry = ProjectRegistry(projectRegistryAddress);
+        projectList = IProjectList(projectRegistry);
+
+        (, address controllerAddress) = createOsmoticControllerAndPoolImpl(
+            VERSION, projectRegistryAddress, MIME_TOKEN_FACTORY_ADDRESS, ROUND_DURATION, CFA_V1_FORWARDER_ADDRESS
+        );
+        controller = OsmoticController(controllerAddress);
+        address mimeTokenAddress = createMimeTokenFromController(controllerAddress, MERKLE_ROOT);
+        mimeToken = MimeToken(mimeTokenAddress);
+
+        _claimMimeTokens(mimeToken);
+
+        pool = OsmoticPool(
+            createOsmoticPoolFromController(
+                controllerAddress, FUNDING_TOKEN_ADDRESS, mimeTokenAddress, address(projectList), OSMOTIC_PARAMS
+            )
+        );
+
+        poolOwner = pool.owner();
+    }
+
+    function _createProject() internal returns (uint256) {
+        uint256 projectId = createProjectInRegistry(projectRegistry);
+
+        projectIds.push(projectId);
+
+        return projectId;
+    }
+
+    function _createProjects(uint256 _numProjects) internal {
+        for (uint256 i = projectIds.length; i < _numProjects; i++) {
+            _createProject();
+        }
+    }
+
+    function _supportProject(address _account, uint256 _projectId, int256 _supportDelta) internal {
+        ProjectSupport[] memory participantSupports = new ProjectSupport[](1);
+        participantSupports[0] = ProjectSupport(_projectId, _supportDelta);
+
+        vm.prank(_account);
+        pool.supportProjects(participantSupports);
+    }
+
+    function _supportAllProjects(address _account, int256 _supportDelta) internal {
+        ProjectSupport[] memory participantSupports = new ProjectSupport[](pool.MAX_ACTIVE_PROJECTS());
+
+        for (uint256 i = 0; i < projectIds.length; i++) {
+            participantSupports[i] = ProjectSupport(projectIds[i], _supportDelta);
+        }
+
+        vm.prank(_account);
+        pool.supportProjects(participantSupports);
+    }
+
+    function _activateAllProjects() internal {
+        for (uint256 i = 0; i < projectIds.length; i++) {
+            pool.activateProject(projectIds[i]);
+        }
+    }
+
+    function _processFuzzedSupports(
```

Comment:
One last thing, I would have liked a bit more guidance to understand this logic. What do you think if we include a few comments?

Reply:
Sure! We can do that